src/dune_rules/pkg_rules.ml

```diff
@@ -318,18 +318,17 @@ module Pkg = struct
   let is_system_provided t = t.system_provided
 
   module Toolchain = struct
-    let is_compiler_and_toolchains_enabled =
-      if Config.get Toolchains_config.enabled
-      then
-        let module Package_name = Dune_pkg.Package_name in
-        let compiler_package_names =
-          Package_name.Set.of_list (* TODO don't hardcode these names here *)
-            [ Package_name.of_string "ocaml-base-compiler"
-            ; Package_name.of_string "ocaml-variants"
-            ]
-        in
-        fun t -> Package_name.Set.mem compiler_package_names t.info.name
-      else Fun.const false
+    let is_compiler_and_toolchains_enabled t =
+      Config.get Toolchains_config.enabled
+      &&
+      let module Package_name = Dune_pkg.Package_name in
+      let compiler_package_names =
```

Comment:
In my original PR this set was computed once ever and then used in each call to `is_compiler_and_toolchains_enabled`. There's no point using a set here if it's being computed on each call - we might as well just linearly search the list.

Reply:
I replaced the use of a set here with a linear search of the list.

Reply:
Sure, that's fine. Note that I only changed your code because it was evaluating the config as a module side effect. That's not allowed because the config needs to be initialized first. You can keep using a set but you need to delay the check with lazy.

In any case, it will not matter as binary search and linear search are pretty close when the size of the collection is 2.